src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -290,14 +290,27 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
                 var clrRundown = new ClrRundownTraceEventParser(source);
                 clrRundown.RuntimeStart += doAtRuntimeStart;
                 source.Clr.RuntimeStart += doAtRuntimeStart;
+                Dictionary<TraceProcess, TraceLoadedDotNetRuntime> processRuntimes = null;
 
                 CircularBuffer<ThreadWorkSpan> RecentThreadSwitches = new CircularBuffer<ThreadWorkSpan>(1000);
                 source.Kernel.ThreadCSwitch += delegate (CSwitchTraceData data)
                 {
+                    if (processRuntimes == null)
+                    {
+                        // lazy initialization
+                        processRuntimes = new Dictionary<TraceProcess, TraceLoadedDotNetRuntime>();
+                        foreach (var proc in source.Processes())
+                        {
+                            var runtime = proc.LoadedDotNetRuntime();
+                            if (runtime != null)
+                                processRuntimes[proc] = runtime;
+                        }
+                    }
+
```

Comment:
The processRuntime cache needs to be built overtime.  The contents of both source.Processes and LoadedDotNetRuntime are built as events are processed.